hikari/impl/rest_bot.py

```diff
@@ -0,0 +1,498 @@
+    def event_factory(self) -> event_factory_.EventFactory:
+        return self._event_factory
+
+    @staticmethod
+    def print_banner(banner: typing.Optional[str], allow_color: bool, force_color: bool) -> None:
+        """Print the banner.
+
+        This allows library vendors to override this behaviour, or choose to
+        inject their own "branding" on top of what hikari provides by default.
+
+        Normal users should not need to invoke this function, and can simply
+        change the `banner` argument passed to the constructor to manipulate
+        what is displayed.
+
+        Parameters
+        ----------
+        banner : typing.Optional[builtins.str]
+            The package to find a `banner.txt` in.
+        allow_color : builtins.bool
+            A flag that allows advising whether to allow color if supported or
+            not. Can be overridden by setting a `"CLICOLOR"` environment
+            variable to a non-`"0"` string.
+        force_color : builtins.bool
+            A flag that allows forcing color to always be output, even if the
+            terminal device may not support it. Setting the `"CLICOLOR_FORCE"`
+            environment variable to a non-`"0"` string will override this.
+
+        !!! note
+            `force_color` will always take precedence over `allow_color`.
+        """
+        ux.print_banner(banner, allow_color, force_color)
+
+    async def close(self) -> None:
+        await self._server.close()
+
+    async def join(self) -> None:
+        await self._server.join()
+
+    async def on_interaction(self, body: bytes, signature: bytes, timestamp: bytes) -> interaction_server_.Response:
+        return await self._server.on_interaction(body, signature, timestamp)
+
+    def run(
+        self,
+        asyncio_debug: typing.Optional[bool] = None,
+        backlog: int = 128,
+        check_for_updates: bool = True,
+        close_loop: bool = True,
+        close_passed_executor: bool = False,
+        coroutine_tracking_depth: typing.Optional[int] = None,
+        enable_signal_handlers: bool = True,
+        host: typing.Optional[typing.Union[str, typing.Sequence[str]]] = None,
+        path: typing.Optional[str] = None,
+        port: typing.Optional[int] = None,
+        reuse_address: typing.Optional[bool] = None,
+        reuse_port: typing.Optional[bool] = None,
+        shutdown_timeout: float = 60.0,
+        socket: typing.Optional[socket_.socket] = None,
+        ssl_context: typing.Optional[ssl.SSLContext] = None,
+    ) -> None:
+        """Open this REST server and block until it closes.
+
+        Other Parameters
+        ----------------
+        asyncio_debug : builtins.bool
+            Defaults to `builtins.False`. If `builtins.True`, then debugging is
+            enabled for the asyncio event loop in use.
+        backlog : builtins.int
+            The number of unaccepted connections that the system will allow before
+            refusing new connections.
```

Comment:
Would this make more sense to rename to `simultaneous_connections`?

Reply:
this isn't about active connections, this about how many unaccepted connections will be queued in a backlog before it starts instantly failing new connections